anvio/scgsdatabase.py

```diff
@@ -0,0 +1,791 @@
+
+
+#!/usr/bin/env python3
+
+import os
+import gzip
+import shutil
+import requests
+from io import BytesIO
+import anvio
+import pickle
+import re
+import shutil
+import subprocess
+import sys
+import tarfile
+import anvio.fastalib as u
+from anvio.drivers.diamond import Diamond
+import anvio.terminal as terminal
+import anvio.pfam as pfam
+import anvio.dbops as dbops
+import anvio.utils as utils
+import anvio.terminal as terminal
+import anvio.filesnpaths as filesnpaths
+
+from anvio.errors import ConfigError
+
+
+run = terminal.Run()
+progress = terminal.Progress()
+pp = terminal.pretty_print
+run_quiet = terminal.Run(log_file_path=None,verbose=False)
+progress_quiet = terminal.Progress(verbose=False)
+
+
+#run_quiet = terminal.Run(verbose=False)
+#progress_quiet = terminal.Progress(verbose=False)
+
+_author__ = "Developers of anvi'o (see AUTHORS.txt)"
+__copyright__ = "Copyleft 2015-2018, the Meren Lab (http://merenlab.org/)"
+__license__ = "GPL 3.0"
+__version__ = anvio.__version__
+__maintainer__ = "Quentin Clayssen"
+__email__ = "quentin.clayssen@gmail.com"
+
+
+run = terminal.Run()
+progress = terminal.Progress()
+pp = terminal.pretty_print
+run_quiet = terminal.Run(verbose=False)
+progress_quiet = terminal.Progress(verbose=False)
+
+
+### open fasta once.... make db of Bacteria_71 fasta .......
+class scgsdatabase():
+    def __init__(self, args, run=run, progress=progress):
+        self.args = args
+        self.run = run
+        self.progress = progress
+
+        A = lambda x: args.__dict__[x] if x in args.__dict__ else None
+
+        self.genes_files_directory = args.genes_files_directory
+
+        self.scgs_directory = args.scgs_directory
+
+        #self.num_threads=args.num_threads
+
+
+
+        self.dictionnary_level=self.dictionnary_level()
+
+
+
+        if not args.taxofiles:
+            self.taxofiles = os.path.join(os.path.dirname(anvio.__file__), 'data/misc/SCG/merge_taxonomy.tsv')
+
+        #if not args.outtsv:
+        self.outtsv = os.path.join(os.path.dirname(anvio.__file__), 'data/misc/SCG/mergedb/matching_taxonomy.tsv')
+
+        #if not args.num_threads:
+        self.num_threads=args.num_threads
+
+        if not args.num_threads:
+            self.num_threads=2
```

Comment:
if there are no threads, then `num_threads` should be 1. what will happen if someone runs 100 of these on a machine with only 100 cores? if there is no parameter, default should always the most conceivable value. in this case, that value is 1.